libsolidity/interface/StandardCompiler.cpp

```diff
@@ -929,6 +929,13 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 				metadataHash == "bzzr1" ?
 				CompilerStack::MetadataHash::Bzzr1 :
 				CompilerStack::MetadataHash::None;
+		if (ret.metadataFormat == CompilerStack::MetadataFormat::NoMetadata && ret.metadataHash != CompilerStack::MetadataHash::None)
+			return formatFatalError(
+				Error::Type::JSONError,
+				"When the parameter \"appendCBOR\" is set to false, the parameter \"bytecodeHash\" cannot be set to \"" +
+				metadataHash +
+				"\". The parameter \"bytecodeHash\" should either be skipped, or set to \"none\"."
```

Comment:
```suggestion
				"\". The parameter \"bytecodeHash\" should either be skipped, or set to \"" + CompilerStack::MetadataHash::None + "\"."
```

Reply:
Ah, wait, we can't do that. Thought for a moment that we have `<<` for this enum but this is `+` and `metadataHash` is just a string.